### PR TITLE
feat: S3 backed recording ingestion (take 2)

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

We had a stab at this before but now it's go-time. This PR is focused _only_ on ingestion - hooking into the existing stream of incoming events and writing them to S3. It purposefully **does not** take care of the retrieval. The goal is to get this in and test in parallel to see if all the data seems to be running correctly before following up with a gradual roll out plan.

## Changes

* Adds a new service to handle ingestion of Session Recording data, buffering to a certain size and flushing to S3

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## TODO
See https://github.com/PostHog/posthog/issues/15200

### CODE REVIEW

File: plugin-server/functional_tests/session-recordings-blob-ingestion.test.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    await Promise.all([await dlqConsumer.disconnect()])
+})
+
+test.skip(`single recording event writes data to local tmp file`, async () => {
+    const teamId = await createTeam(organizationId)
+    const distinctId = new UUIDT().toString()
+    const uuid = new UUIDT().toString()
+    const sessionId = new UUIDT().toString()
+    const veryLongString = generateVeryLongString()
+    await capture({
```
Comment (hazzadous) [inline plugin-server/functional_tests/session-recordings-blob-ingestion.test.ts:64]: fwiw I think you'll want to specify the topic to be the `session_recording_events` topic here. For backwards compat. we're forwarding these events from the `events_plugin_ingestion` topic but we should remove this.

File: plugin-server/src/kafka/consumer.ts [hunk 2]
```diff
@@ -160,10 +171,10 @@ export const disconnectConsumer = async (consumer: RdKafkaConsumer) => {
     await new Promise((resolve, reject) => {
         consumer.disconnect((error, data) => {
             if (error) {
-                status.error('🔥', 'Failed to disconnect session recordings consumer', { error })
+                status.error('🔥', 'Failed to disconnect noderd-kafka consumer', { error })
```
Comment (hazzadous) [inline plugin-server/src/kafka/consumer.ts:174]:
```suggestion
                status.error('🔥', 'Failed to disconnect node-rdkafka consumer', { error })
```

File: plugin-server/src/kafka/consumer.ts [hunk 3]
```diff
@@ -160,10 +171,10 @@ export const disconnectConsumer = async (consumer: RdKafkaConsumer) => {
...
                 reject(error)
             } else {
-                status.info('🔁', 'Disconnected session recordings consumer')
+                status.info('🔁', 'Disconnected session noderd-kafka consumer')
```
Comment (hazzadous) [inline plugin-server/src/kafka/consumer.ts:177]:
```suggestion
                status.info('🔁', 'Disconnected session node-rdkafka consumer')
```

File: plugin-server/functional_tests/session-recordings-blob-ingestion.test.ts (hunk 1 above)
Comment (pauldambra) [reply plugin-server/functional_tests/session-recordings-blob-ingestion.test.ts:64]: 🤔

**Pull request merged**